src/backend/routers/iep.ts

```diff
@@ -30,6 +31,45 @@ export const iep = router({
       return result;
     }),
 
+  editGoal: authenticatedProcedure
+    .input(
+      z.object({
+        goal_id: z.string(),
+        description: z.string(),
+      })
+    )
+    .mutation(async (req) => {
+      const { goal_id, description } = req.input;
+      const { userId } = req.ctx.auth;
+
+      // make sure that this goal belongs to this case manager
+      const authCheck = await req.ctx.db
+        .selectFrom("goal")
+        .fullJoin("iep", "iep.iep_id", "goal.goal_id")
+        .fullJoin("student", "student.student_id", "iep.student_id")
+        .where("student.assigned_case_manager_id", "=", userId)
```

Comment:
This is actually a somewhat complicated issue to address and I expect will require a larger conversation, but the check in the editGoal isn't actually redundant as things currently stand. I should say first of all that I don't know to what extent tRPC protects these backend procedures against exposed api endpoints by keeping it so that the only way that a route can be hit is through a frontend method calling the tRPC procedure. If tRPC does that, there may be an ability to eliminate some redundancy wherever there is a place that we can trust that the frontend is correctly narrowing the scope of what a user can see.

Having said that, the editGoal procedure appears on the frontend on the [goal_id] page and is loaded by the getGoal procedure, which I wrote based on the getGoals procedure right above it. As it currently stands, none of the procedures in the iep router (including getGoals and getGoal) do a check on the assigned case manager, except for editGoal. I haven't been able to directly check this, but what I believe that this means is that any user who passes the authentication check would be able to see every level of detail for a student, even if that student is not assigned to that user. All a user would need would be the id of the student or goal to go to that page of the app. I didn't want to blow the scope of my issue up even more than what it was currently at, so I figured I would add this check to the editGoal procedure and then perhaps a new issue can be made later if a decision is made regarding any additional permission checks that need to be made.